data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/GenericExpressionEvaluatorTest.java

```diff
@@ -142,5 +150,47 @@ void isValidFormatExpressionsReturnsFalseWhenIsValidKeyAndValidExpressionIsFalse
         assertThat(statementEvaluator.isValidFormatExpression(format), equalTo(false));
     }
 
+    @ParameterizedTest
+    @ArgumentsSource(FormatExpressionsToExtractedDynamicKeysArgumentProvider.class)
+    void extractDynamicKeysFromFormatExpression_returns_expected_result(final String formatExpression, final List<String> expectedDynamicKeys) {
+        final List<String> result = statementEvaluator.extractDynamicKeysFromFormatExpression(formatExpression);
+
+        assertThat(result, notNullValue());
+        assertThat(result.equals(expectedDynamicKeys), equalTo(true));
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(FormatExpressionsToExtractedDynamicExpressionsArgumentProvider.class)
+    void extractDynamicExpressionsFromFormatExpression_returns_expected_result(final String formatExpression, final List<String> expectedDynamicExpressions) {
```

Comment:
can we add a negative test where invalid expression is passed ?

Reply:
There is a negative case where invalid expression syntax is found, it will just not be added to the list. This method is just supposed to parse out valid existing expressions